validator/client/validator.go

```diff
@@ -1242,25 +1241,41 @@ func (v *validator) EventStreamIsRunning() bool {
 	return v.validatorClient.EventStreamIsRunning()
 }
 
-func (v *validator) HealthTracker() health.Tracker {
-	return v.nodeClient.HealthTracker()
-}
-
 func (v *validator) Host() string {
 	return v.validatorClient.Host()
 }
 
 func (v *validator) ChangeHost() {
-	if len(v.beaconNodeHosts) == 1 {
-		log.Infof("Beacon node at %s is not responding, no backup node configured", v.Host())
-		return
-	}
 	next := (v.currentHostIndex + 1) % uint64(len(v.beaconNodeHosts))
-	log.Infof("Beacon node at %s is not responding, switching to %s...", v.beaconNodeHosts[v.currentHostIndex], v.beaconNodeHosts[next])
+	log.WithFields(logrus.Fields{
+		"current_host": v.beaconNodeHosts[v.currentHostIndex],
+		"next_host":    v.beaconNodeHosts[next],
+	}).Warn("Beacon node is not responding, switching to host")
 	v.validatorClient.SetHost(v.beaconNodeHosts[next])
 	v.currentHostIndex = next
 }
 
+func (v *validator) FindHealthyHost(ctx context.Context) bool {
+	// Tail-recursive closure keeps retry count private.
+	var check func(remaining int) bool
+	check = func(remaining int) bool {
+		if v.nodeClient.IsHealthy(ctx) { // healthy → done
+			return true
+		}
+		if len(v.beaconNodeHosts) == 1 {
+			log.WithField("host", v.Host()).Debug("Beacon node is not responding, no backup node configured")
```

Comment:
```suggestion
			log.WithField("host", v.Host()).Info("Beacon node is not responding, no backup node configured")
```
because the switching log above is info level